src/components/VariantSelector/VariantSelector.ts

```diff
@@ -0,0 +1,181 @@
+ * customized using CSS custom properties.
+ *
+ * @property {string} handle - The Shopify product handle to fetch data for. Required.
+ *
+ * @fires variantchange - Emitted when variant selection changes, contains { variant, product }
+ *
+ * @example
+ * ```html
+ * <nosto-variant-selector handle="awesome-product"></nosto-variant-selector>
+ * ```
+ */
+@customElement("nosto-variant-selector", { observe: true })
+export class VariantSelector extends NostoElement {
+  /** @private */
+  static attributes = {
+    handle: String
+  }
+
+  handle!: string
+
+  /** Internal state for current selections */
+  selectedOptions: Record<string, string> = {}
+
+  constructor() {
+    super()
+    this.attachShadow({ mode: "open" })
+  }
+
+  async attributeChangedCallback() {
+    if (this.isConnected) {
+      await loadAndRenderMarkup(this)
+    }
+  }
+
+  async connectedCallback() {
+    assertRequired(this, "handle")
+    await loadAndRenderMarkup(this)
+  }
+}
+
+async function loadAndRenderMarkup(element: VariantSelector) {
+  element.toggleAttribute("loading", true)
+  try {
+    const productData = await fetchProductData(element.handle)
+
+    // Initialize selections with first value of each option
+    initializeDefaultSelections(element, productData)
+
+    const selectorHTML = generateVariantSelectorHTML(element, productData)
+
+    // Use constructible stylesheets if supported, fallback to inline styles
+    if ("adoptedStyleSheets" in element.shadowRoot!) {
+      if (!cachedStyleSheet) {
+        cachedStyleSheet = new CSSStyleSheet()
+        await cachedStyleSheet.replace(variantSelectorStyles)
+      }
+      element.shadowRoot!.adoptedStyleSheets = [cachedStyleSheet]
+      element.shadowRoot!.innerHTML = selectorHTML.html
+    } else {
+      element.shadowRoot!.innerHTML = `
+        <style>${variantSelectorStyles}</style>
+        ${selectorHTML.html}
+      `
+    }
+
+    // Setup event listeners for option buttons
+    setupOptionListeners(element)
+
+    updateActiveStates(element)
+    emitVariantChange(element, productData)
+  } finally {
+    element.toggleAttribute("loading", false)
+  }
+}
+
+function initializeDefaultSelections(element: VariantSelector, product: ShopifyProduct) {
+  product.options.forEach(option => {
+    if (option.values.length > 0) {
+      element.selectedOptions[option.name] = option.values[0]
+    }
+  })
+}
+
+function setupOptionListeners(element: VariantSelector) {
+  if (!element.shadowRoot) return
+
+  element.shadowRoot.addEventListener("click", async e => {
+    const target = e.target as HTMLElement
+    if (target.classList.contains("variant-option-value")) {
+      e.preventDefault()
+      const optionName = target.getAttribute("data-option-name")
+      const optionValue = target.getAttribute("data-option-value")
+
+      if (optionName && optionValue) {
+        await selectOption(element, optionName, optionValue)
```

Comment:
Skip the event where same option is clicked again?

Reply:
done